backend/app.js

```diff
@@ -22,16 +51,84 @@ const db = {
   ]
 }
 
+//Allow app to use CORS. Important for transferring JSON files.
 app.use(cors())
 
+//Set appropriate headers for server
+app.use(function(req, res, next){
+    res.setHeader('Access-Control-Allow-Origin', 'http://localhost:4200');
+    res.header("Access-Control-Allow-Headers", "Origin, X-Requested0With, Content-Type, Accept");
+    res.header('Access-Control-Allow-Methods', 'POST, GET, PUT, DELETE, OPTIONS');
+    next();
+});
+
+//Support parsing of application/json type post data
+app.use(bodyParser.json());
+
+app.use(bodyParser.urlencoded({extended: true}));
+
+//Tell mongoose to connect to MongoDB instance and look for ember data.
+//If no database exists, it will create one for us.
+mongoose.connect('mongodb://localhost/emberData');
```

Comment:
Once we get to that point, we can parameterise the location of mongo i.e. The local host part